src/hash.h

```diff
@@ -4,38 +4,38 @@
  * This file is part of libgit2, distributed under the GNU GPL v2 with
  * a Linking Exception. For full terms see the included COPYING file.
  */
+
 #ifndef INCLUDE_hash_h__
 #define INCLUDE_hash_h__
 
 #include "common.h"
 
 #include "git2/oid.h"
 
-typedef struct git_hash_prov git_hash_prov;
-typedef struct git_hash_ctx git_hash_ctx;
-
-int git_hash_ctx_init(git_hash_ctx *ctx);
-void git_hash_ctx_cleanup(git_hash_ctx *ctx);
-
-#if defined(GIT_SHA1_COLLISIONDETECT)
-# include "hash/hash_collisiondetect.h"
-#elif defined(GIT_SHA1_COMMON_CRYPTO)
-# include "hash/hash_common_crypto.h"
-#elif defined(GIT_SHA1_OPENSSL)
-# include "hash/hash_openssl.h"
-#elif defined(GIT_SHA1_WIN32)
-# include "hash/hash_win32.h"
-#elif defined(GIT_SHA1_MBEDTLS)
-# include "hash/hash_mbedtls.h"
-#else
-# include "hash/hash_generic.h"
-#endif
-
 typedef struct {
 	void *data;
 	size_t len;
 } git_buf_vec;
 
+typedef enum {
+	GIT_HASH_ALGO_UNDEF = 0,
```

Comment:
Nitpick: `UNKNOWN` ? It's never really undefined, as it's either SHA1 or something we don't know. Semantics 😉.

Reply:
Agreed